backend/src/main/java/com/example/backend/use_case/route_details/RouteDetailsImplementation.java

```diff
@@ -0,0 +1,42 @@
+package com.example.backend.use_case.route_details;
+
+import com.example.backend.data_access.route.RouteDataAccessInterface;
+import com.example.backend.entity.Route;
+import com.example.backend.entity.RouteDirection;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class RouteDetailsImplementation implements RouteDetailsService {
+    @Autowired
+    private RouteDataAccessInterface routeDAO;
+
+    /** Returns the route details for the given routeTag and dirTag. Returns null if the routeTag or dirTag is invalid.
+     *
+     * @param routeTag String representing the routeTag
+     * @param dirTag String representing the dirTag
+     * @return RouteDetailsOutputData object containing the routeTag, dirTag, dirName, and routeShape. Null if the routeTag or dirTag is invalid.
+     */
+    public RouteDetailsOutputData execute(String routeTag, String dirTag) {
+        Route route = routeDAO.getRouteByRouteTag(routeTag);
+
+        // TODO: Fix error handling
+        if (route == null) {
+            return null;
+        }
+
+        RouteDirection routeDirection = route.getRouteDirections().get(dirTag);
+
+        if (routeDirection == null) {
+            return null;
+        }
+
+        return new RouteDetailsOutputData(
+                routeTag,
+                dirTag,
+                routeDirection.getName(),
+                routeDAO.getRouteShapes().get(routeTag)
```

Comment:
since StopDetails is going to pass in direction names along with direction tags, is this still needed? the focus should just be returning shapes right

Reply:
I mean, imo it should still be returned. We don't know what the front end has access to, so this use case focuses specifically on returning details about any given route and direction.

Reply:
I mean I feel like it's not *needed*, but also it doesn't break anything